adal/tests/Test.ADAL.NET.UIAutomation/XamarinUITestController.cs

```diff
@@ -0,0 +1,118 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Test.ADAL.NET.UIAutomation.Infrastructure;
+using Xamarin.UITest;
+
+namespace Test.ADAL.NET.UIAutomation
+{
+    public class XamarinUITestController : ITestController
+    {
+        IApp app;
+        TimeSpan defaultSearchTimeout;
+        TimeSpan defaultRetryFrequency;
+        TimeSpan defaultPostTimeout;
+        const int defaultSearchTimeoutSec = 10;
+        const int defaultRetryFrequencySec = 1;
+        const int defaultPostTimeoutSec = 1;
+        const string CSSIDSelector = "[id|={0}]";
+        private readonly ILabService _labService;
+
+        public XamarinUITestController(IApp app)
+        {
+            this.app = app;
+            this.defaultSearchTimeout = new TimeSpan(0, 0, defaultSearchTimeoutSec);
+            this.defaultRetryFrequency = new TimeSpan(0, 0, defaultRetryFrequencySec);
+            this.defaultPostTimeout = new TimeSpan(0, 0, defaultPostTimeoutSec);
+            _labService = new LabServiceApi();
+        }
+
+        public void Tap(string elementID)
+        {
+            Tap(elementID, false, defaultSearchTimeout);
+        }
+
+        public void Tap(string elementID, bool isWebElement)
+        {
+            Tap(elementID, isWebElement, defaultSearchTimeout);
+        }
+
+        public void Tap(string elementID, int waitTime, bool isWebElement)
+        {
+            Tap(elementID, isWebElement, new TimeSpan(0, 0, waitTime));
+        }
+
+        public void EnterText(string elementID, string text, bool isWebElement)
```

Comment:
>EnterText [](start = 20, length = 9)

I mentioned this in another place. For some of these text entering elements the automation takes forever. does something better exist in the test infra, to have the text inputs go faster (e.g. via copy/paste).

Reply:
@henrik-me  The good thing about the Xamarin UI test framework is that it does a copy and paste for text input by default when using "EnterText()".

Reply:
Can the same be used for the desktop tests?

Reply:
With the current setup of the desktop tests, slow text input is the only option. However, with newer versions of the selenium framework, it may now be possible. ill investigate.